### DISCUSSION TITLE
Simplesamlphp Proxy -> Azure IdP

### DISCUSSION THREAD

**Discussion by alexbenisch** [bergie/passport-saml#663] [Q&A]
Hi,
I am trying to connect to a Simplesamlphp Proxy in front of an IdP on the Azure Cloud with passport-saml, although I am not sure if passport-adfs would be more suitable.
There is neither access to the simplesamlphp Proxy logs nor to the logs of the IdP itself.
Hints about the config.js, metadata and route.js would be greatly appreciated.
I have tried every logical combination of options in the reference and I would greatly appreciate any form of advice about the debugging, I use the Chrome extension Saml tracer right now. Also I used samltest.id to configure a reference implementation, that seemed to have worked. 
The generateServiceProviderMetadata function is another part where I am stuck right now. At the moment I serve the metadata XML file generated with samltool.com because I can't get the function to include the sign cert, only the encryption certificate is included.
The IdP admin can't be of any help as he does not know a lot about Saml and the configuration options.

Looking forward to any input
Thanks 
Alexander

**Comment by cjbarth** [comment]
It would be helpful if you included some error, or sample code, or something more concrete. You should be able to capture any errors or even step-debug through `passport-saml` to get more information even if you don't have access to server logs.
